### PR TITLE
Stack based build workflow

### PR DESCRIPTION
**Pull request by blackgnezdo**
Based on @awpr's hs-portray workflow.

### CODE REVIEW

**Comment by blackgnezdo** [comment]
The workflows seem to be functional: https://github.com/blackgnezdo/proto-lens/actions/runs/1276245719

I'm not sure what's amiss to make them trigger for this repository. Maybe merging into master is a prerequisite?

**Comment by blackgnezdo** [comment]
BTW, caching works as rebuild took merely 5 minutes whereas clean builds required 30:
https://github.com/blackgnezdo/proto-lens/runs/3715054683?check_suite_focus=true

**Review by jinwoo — APPROVED** [review]
Thanks!

File: stack-8.10.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
-resolver: ghc-8.10.1
+resolver: lts-18.10
```
Comment (jinwoo) [inline stack-8.10.yaml:7]: You changed this to lts-18.6 in https://github.com/google/proto-lens/pull/417, and now it's lts-18.10. Is lts-18.10 correct?

Comment (blackgnezdo) [reply stack-8.10.yaml:7]: Yeah, again #417 was pending long enough that I forgot about it and move to a newer lts here.

**Comment by awpr** [comment]
IIRC newly-added workflows don't trigger until they're first merged, so they should start happening once this is merged and on subsequent PRs.

**Review by jinwoo — APPROVED** [review]

**Pull request merged**